### PR TITLE
Fix: Contacts can now be searched with a leading @ again.

### PR DESCRIPTION
**Pull request by annando**
This fixes the issue that it wasn't possible anymore to search for contact when you used a leading `@`.

### CODE REVIEW

**Review by MrPetovan — APPROVED** [review]

File: src/Module/BaseSearch.php [hunk 1]
```diff
@@ -91,14 +91,16 @@ public static function performContactSearch(string $search, string $prefix = '')
...
+			if ($localSearch) {
+				$pager->setItemsPerPage(80);
+				$results = Search::getContactsFromLocalDirectory($search, $type, $pager->getStart(), $pager->getItemsPerPage());
+			} elseif (Search::getGlobalDirectory()) {
+				$results = Search::getContactsFromGlobalDirectory($search, $type, $pager->getPage());
+				$pager->setItemsPerPage($results->getItemsPage());
+			} else {
+				$results = new ResultList();
+			}
 		}
```
Comment (MrPetovan) [inline src/Module/BaseSearch.php]: I've been able to fix the same issue with just the following change with no changes to `src/Core/Search.php`.
```suggestion
		if (empty($results) || !$results->getTotal()) {
			if ($localSearch) {
				$pager->setItemsPerPage(80);
				$results = Search::getContactsFromLocalDirectory($search, $type, $pager->getStart(), $pager->getItemsPerPage());
			} elseif (Search::getGlobalDirectory()) {
				$results = Search::getContactsFromGlobalDirectory($search, $type, $pager->getPage());
				$pager->setItemsPerPage($results->getItemsPage());
			} else {
				$results = new ResultList();
			}
		}
```
What's the purpose of the changes to `Core\Search::getContactsFromProbe` except return `null` to satisfy the `empty($results)` condition?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/BaseSearch.php (hunk 1 above)
Comment (annando) [reply src/Module/BaseSearch.php]: The used array fields there should always be filled. If not then there is some deeper issue.

Comment (MrPetovan) [reply src/Module/BaseSearch.php]: Why?

Comment (annando) [reply src/Module/BaseSearch.php]: Either all fields are there or none. The latter case is handled with the `empty(...)` check. If we experience any issue with some missing fields, then something really is wrong - and we should know about it. Each little warning/notice that we mark with `?? ...` is a potential "let's put a sticker over the warning light" thing.

Comment (MrPetovan) [reply src/Module/BaseSearch.php]: What fields are you talking about?

Comment (nupplaphil) [reply src/Module/BaseSearch.php]: And what about adding an exception? This is a REAL notice instead of unexpected PHP warnings.

File: src/Core/Search.php [hunk 2]
```diff
@@ -129,8 +131,8 @@ public static function getContactsFromGlobalDirectory(string $search, int $type
 
 		$resultList = new ResultList(
 			($results['page']         ?? 0) ?: 1,
-			 $results['count']        ?? 0,
-			($results['itemsperpage'] ?? 0) ?: 30
+			($results['itemsperpage'] ?? 0) ?: 30,
+			$results['count']        ?? 0
```
Comment (MrPetovan) [inline src/Core/Search.php]: Why did you swap these? The parameter order is `page, count, itemsperpage`.
```suggestion
			$results['count']         ?? 0,
			($results['itemsperpage'] ?? 0) ?: 30
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Search.php (hunk 2 above)
Comment (annando) [reply src/Core/Search.php]: You re right.

**Review by MrPetovan — APPROVED** [review]
